tdms/tests/system/data/input_generation/generate_test_input.py

```diff
@@ -0,0 +1,242 @@
+import os
+from glob import glob
+from pathlib import Path
+from subprocess import PIPE, Popen
+from typing import Union
+
+import yaml
+
+LOCATION_OF_THIS_FILE = os.path.dirname(os.path.abspath(__file__))
+# MATLAB executable path
+MATLAB_PATH = "matlab"
+# Additional options for running matlab on the command-line
+MATLAB_OPTS = ["-nodisplay", "-nodesktop", "-nosplash", "-r"]
+# Paths to matlab functions not in LOCATION_OF_THIS_FILE
+MATLAB_EXTRA_PATHS = [
+    os.path.abspath(LOCATION_OF_THIS_FILE + "/bscan"),
+    os.path.abspath(LOCATION_OF_THIS_FILE + "/matlab"),
+]
+
+
+class BScanArguments:
+    """Class to tidily contain the input arguments that need to be passed into the MATLAB function of the same name, to generate the input data.
+
+    The bscan/run_bscan.m file contains the matlab function which generates the input data. Regrettably, we need to specify particular inputs to this script for each test, which requires us to translate the argument values as read from the config.yaml file into a long string of values in the correct order, which can in turn be called from matlab.
+
+    BScanArguments is essentially a glorified dictionary, it's members sharing the names of the input arguments to run_bscan. It's create_bscan_argument can be used to convert the values that need to be passed into a string of the form:
+    run_bscan(input_arguments_in_the_correct_order).
```

Comment:
How does `black` allow this? I'm expecting a [forced wrap at 88](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length).